src/main/java/de/ait/secondlife/services/OfferServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+    @Override
+    public OfferResponseDto findOfferById(UUID id) {
+        if (id == null) throw new IdIsNullException();
+        Offer offer = offerRepository.findByIdAndIsActiveTrue(id)
+                .orElseThrow(() -> new OfferNotFoundException(id));
+        return mappingService.toRequestDto(offer);
+    }
+
+    @Override
+    public OfferResponseWithPaginationDto findOffersByUserId(Long id, Pageable pageable) {
+        if (id == null) throw new IdIsNullException();
+        Page<Offer> pageOfOffer = offerRepository.findByUserIdAndIsActiveTrue(id, pageable);
+        return offersToOfferRequestWithPaginationDto(pageOfOffer);
+    }
+
+    @Override
+    public OfferResponseDto createOffer(OfferCreationDto dto) {
+        try {
+            Offer newOffer = mappingService.toOffer(dto);
+            newOffer.setId(null);
+            newOffer.setStatus(statusSevice.getStatusByName(OfferStatus.DRAFT.name()));
+            newOffer.setAuctionDurationDays(newOffer.getAuctionDurationDays() <= 0 ? 3 : newOffer.getAuctionDurationDays());
+            if (dto.getIsFree() == null) {
+                newOffer.setIsFree(dto.getStartPrice() == null || dto.getStartPrice().compareTo(BigDecimal.ZERO) == 0);
+            }
+            if (Boolean.TRUE.equals(newOffer.getIsFree())) {
+                checkOfferIfIsFree(dto.getStartPrice(), dto.getStep(), dto.getWinBid());
+            } else {
+                if (dto.getStartPrice() == null || dto.getStartPrice().compareTo(BigDecimal.ZERO) == 0)
+                    throw new WrongAuctionParameterException("start prise");
+                if (dto.getStep() == null || dto.getStep().compareTo(BigDecimal.ZERO) == 0)
+                    throw new WrongAuctionParameterException("step");
+                if (dto.getWinBid() != null && dto.getWinBid().compareTo(BigDecimal.ZERO) == 0) {
+                    throw new WrongAuctionParameterException("winBid");
+                }
+            }
+            newOffer = offerRepository.save(newOffer);
+            return mappingService.toRequestDto(newOffer);
+        } catch (ConstraintViolationException | DataIntegrityViolationException e) {
+            throw new CreateOfferConstraintViolationException("Constraint violation: " + e.getMessage());
+        }
+    }
+
+    @Transactional
+    @Override
+    public void updateOffer(OfferUpdateDto dto) {
+        Offer offer = offerRepository.findById(dto.getId())
+                .orElseThrow(() -> new OfferNotFoundException(dto.getId()));
+
+        offer.setTitle(dto.getTitle() == null ? offer.getTitle() : dto.getTitle());
+        offer.setDescription(dto.getDescription() == null ? offer.getDescription() : dto.getDescription());
+        offer.setAuctionDurationDays(dto.getAuctionDurationDays() == null || dto.getAuctionDurationDays() <= 0 ?
+                offer.getAuctionDurationDays() : dto.getAuctionDurationDays());
+        offer.setIsFree(dto.getIsFree() == null ? offer.getIsFree() : dto.getIsFree());
```

Comment:
unnecessary check, since there is a limitation in the DTO
_@NotNull(message = "Is free cannot be null")
private Boolean isFree;_